src/main/java/entities/StateTracker.java

```diff
@@ -160,5 +162,33 @@ public void removeTrackedCourse(String courseId) {
         this.trackedCourses.remove(courseId);
     }
 
+    /**
+     * Check if a document is already downloaded locally
+     * @param documentId the document ID being checked
+     * @return whether the document is downloaded locally
+     */
+    public boolean checkIfDocumentDownloaded(String documentId) {
+        return this.downloadedDocumentPaths.containsKey(documentId);
+    }
+
+    /**
+     * Check if a document is already downloaded locally
```

Comment:
This seems to be the wrong description for `updateDownloadedDocuments`

Reply:
Thanks for highlighting, it should be corrected now